sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -70,6 +70,11 @@ def add_span_processor(span_processor)
         @span_processors << span_processor
       end
 
+      def detect_resources(resource = nil)
```

Comment:
Why pass in this state? Feels like you could `c.resource = auto_detect_resource.merge(service_resource(service_name))` or something similar.

Reply:
Yeah I wasn't too sure what the interface should look like yet.  I like this idea better than what I had.  Updated.